src/main/java/core/basesyntax/strategy/impl/FruitHandlerStrategyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.FruitHandler;
+import core.basesyntax.strategy.FruitHandlerStrategy;
+import java.util.Map;
+
+public class FruitHandlerStrategyImpl implements FruitHandlerStrategy {
+    private Map<FruitTransaction.Operation, FruitHandler> strategies
+            = Map.of(FruitTransaction.Operation.BALANCE, new BalanceImpl(),
+            FruitTransaction.Operation.PURCHASE, new PurchaseImpl(),
+            FruitTransaction.Operation.RETURN, new ReturnImpl(),
+            FruitTransaction.Operation.SUPPLY, new SupplyImpl());
```

Comment:
I'd recommend initializing this map somewhere outside (for example method main) and then passing it using a constructor to this class